op-challenger/game/fault/trace/cannon/state_converter_test.go

```diff
@@ -45,34 +45,42 @@ func TestLoadState(t *testing.T) {
 		state, err := parseState(path)
 		require.NoError(t, err)
 
-		var expected singlethreaded.State
-		require.NoError(t, json.Unmarshal(testState, &expected))
-		require.Equal(t, &expected, state)
+		expected := loadExpectedState(t)
+		require.Equal(t, expected, state)
 	})
 
 	t.Run("Binary", func(t *testing.T) {
-		var expected singlethreaded.State
-		require.NoError(t, json.Unmarshal(testState, &expected))
+		expected := loadExpectedState(t)
 
-		dir := t.TempDir()
-		path := filepath.Join(dir, "state.bin")
-		require.NoError(t, serialize.Write[*singlethreaded.State](path, &expected, 0644))
+		path := writeState(t, "state.bin", expected)
 
 		state, err := parseState(path)
 		require.NoError(t, err)
-		require.Equal(t, &expected, state)
+		require.Equal(t, expected, state)
 	})
 
 	t.Run("BinaryGzip", func(t *testing.T) {
-		var expected singlethreaded.State
-		require.NoError(t, json.Unmarshal(testState, &expected))
+		expected := loadExpectedState(t)
 
-		dir := t.TempDir()
-		path := filepath.Join(dir, "state.bin.gz")
-		require.NoError(t, serialize.Write[*singlethreaded.State](path, &expected, 0644))
+		path := writeState(t, "state.bin.gz", expected)
 
 		state, err := parseState(path)
 		require.NoError(t, err)
-		require.Equal(t, &expected, state)
+		require.Equal(t, expected, state)
 	})
 }
+
+func writeState(t *testing.T, filename string, state *versions.VersionedState) string {
+	dir := t.TempDir()
+	path := filepath.Join(dir, filename)
+	require.NoError(t, serialize.Write(path, state, 0644))
+	return path
+}
+
+func loadExpectedState(t *testing.T) *versions.VersionedState {
```

Comment:
(optional) Maybe worth versioning the tests to run against a couple different state versions.

Reply:
Updated it to use both single and multithreaded.  I've done it by generating the states in the test since we've already got a dependency on cannon. That means the states are mostly default values but saves having to check in an opaque binary file. If it wasn't reusing the code from cannon I think we might want more coverage but I think this should pick up any errors we're likely to introduce (plus we get coverage via the e2e tests as well).